### ISSUE TITLE
Error printed when running flutter_gallery

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#11965]
Running the flutter_gallery with `flutter run` prints the following error, but then continues as expected:

```
Error detected in pubspec.yaml:
#/properties/flutter: unallowed additional property plugin
```

**Comment by jason-simmons** [comment]
This is happening in the AssetBundle builder when it tries to parse the pubspec.yaml manifest for the url_launcher plugin.

AssetBundle.build is trying to validate the plugin's pubspec.yaml against the schema used for an application's pubspec.yaml.  But plugins seem to be using a different format for their manifest.

@szakarias Do the tools need a different schema for plugins?  Or should the _validatePackageManifest call be skipped for a plugin?

**Comment by szakarias** [comment]
I think we can maybe skip the validation step. I'll look into this.

**Comment by Hixie** [comment]
We shouldn't skip it, we should just use the right schema.

**Comment by szakarias** [comment]
Agree, it should be validated, but I was wondering whether we do or should validate it somewhere else. Doesn't it seem a bit strange to validate the plugin registration section while handling assets?

**Comment by Hixie** [comment]
I'm not super familiar with that part of the code (cc @cbracken or @tvolkert who may know it better), but as a general rule I would encourage you to embrace the yak shave and feel absolutely empowered to refactor the code to make it make more sense. :-)

**Comment by szakarias** [comment]
Thanks, that was the plan :-)

**Comment by szakarias** [comment]
Fixed in #12013
